commons/src/config/output.rs

```diff
@@ -0,0 +1,150 @@
+    #[arg(long, alias = "overwrite-out-files")]
+    pub overwrite: bool,
+}
+
+impl ConfigFns for Output {
+    fn merge(&mut self, other: Self) {
+        self.file.replace_none(other.file);
+        self.formats.extend(other.formats.iter());
+        self.format_specific.merge(other.format_specific);
+    }
+
+    fn validate(&self) -> Result<(), logid::log_id::LogId> {
+        self.format_specific.validate()?;
+
+        if self.formats.is_empty() {
+            return Err(set_event_with!(
+                ConfigErrLogId::InvalidConfig,
+                &COMMONS_LOG_ID_MAP,
+                "No output format was set."
+            )
+            .into());
+        }
+
+        if let Some(ref file) = self.file {
+            let mut filepath = file.clone();
+            for format in &self.formats {
+                filepath.set_extension(format.extension());
+
+                if filepath.exists() && !self.overwrite {
+                    return Err(set_event_with!(
+                        ConfigErrLogId::InvalidConfig,
+                        &COMMONS_LOG_ID_MAP,
+                        &format!(
+                            "Output file '{:?}' already exists, but `overwrite` was not set.",
+                            filepath
+                        )
+                    )
+                    .into());
+                }
+            }
+        }
+        Ok(())
+    }
+}
+
+#[derive(
+    Default,
+    Debug,
+    Copy,
+    Clone,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    ValueEnum,
+    Hash,
+    Serialize,
+    Deserialize,
+)]
+pub enum OutputFormat {
+    #[default]
+    Html,
+}
+
+impl OutputFormat {
+    /// Returns the associated file extension for an output format.
+    pub fn extension(&self) -> &str {
+        match self {
+            OutputFormat::Html => "html",
+        }
+    }
+}
+
+impl FromStr for OutputFormat {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "html" => Ok(OutputFormat::Html),
+            o => Err(format!("Bad output format: {}", o)),
+        }
+    }
+}
+
+#[derive(Args, Default, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct OutputFormatSpecific {
+    #[command(flatten)]
+    #[serde(flatten)]
+    pub html: HtmlSpecific,
+}
+
+impl ConfigFns for OutputFormatSpecific {
+    fn merge(&mut self, other: Self) {
+        self.html.merge(other.html);
+    }
+
+    fn validate(&self) -> Result<(), logid::log_id::LogId> {
+        self.html.validate()
+    }
+}
+
+#[derive(Args, Default, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct HtmlSpecific {
+    #[arg(long, value_parser = parse_to_hashset::<PathBuf>, required = false, default_value = "")]
+    pub favicons: HashSet<PathBuf>,
+    #[arg(long, value_parser = parse_to_hashset::<String>, required = false, default_value = "")]
+    pub keywords: HashSet<String>,
+}
```

Comment:
I'm curious, are these options always available, or is there a way to make them depend on other options? For example, make them only available if users specify `html` as one of the output formats. 

It doesn't hurt if they're available always, they won't do anything if html won't be rendered, it's just a thought 😄

Reply:
Clap allows conditional arguments, so maybe it is possible to require `format=html` for these specific options.
I will take a look.

Reply:
After going through the documentation of clap, it seems not possible to enforce a specific value of another argument.
[require_if](https://docs.rs/clap/latest/clap/struct.Arg.html#method.requires_if) only checks the value of the argument itself.

For now lets, allow the two arguments for any output format, but internally only consider them for the `html` output.
Maybe clap will add options to handle this use case at some point.